packages/core/src/editor/BlockNoteEditor.ts

```diff
@@ -726,9 +732,117 @@ export class BlockNoteEditor<
     this.emit("create");
   }
 
-  dispatch = (tr: Transaction) => {
+  /**
+   * Dispatch a ProseMirror transaction.
+   *
+   * @example
+   * ```ts
+   * const tr = editor.transaction;
+   * tr.insertText("Hello, world!");
+   * editor.dispatch(tr);
+   * ```
+   */
+  public dispatch(tr: Transaction) {
+    if (this.transactionState) {
+      const { state, transactions } =
+        this.transactionState.applyTransaction(tr);
```

Comment:
this would apply plugins, right? 
- do we want this?
- Should we check whether all plugins are compatible with this? i.e.: the state that they get will be different from the "actual editor state" (as the actual editor - or at least, the view - still has an old state)

Reply:
This would apply the `filterTransaction` and `appendTransaction` events.
Both of these have function signatures provide the current editorState as an argument [filterTransaction](https://prosemirror.net/docs/ref/#state.PluginSpec.filterTransaction) and [appendTransaction](https://prosemirror.net/docs/ref/#state.PluginSpec.appendTransaction). So operating on the editor state they were given is what they should be doing anyway (since they are run during state transitions), it would be incorrect for them to be pulling anything from `editor.prosemirrorState`, `editor._tiptapEditor.state` or `editor._tiptapEditor.view.state`.

I've checked our plugins and this should not be an issue, I did not have to change any of the plugins for this.

Reply:
We also need the plugins to apply because after an `editor.dispatch` the plugins would normally be assumed to have already run, and we don't want to break this assumption just because the parent called the code within a transaction.